woocommerce/woocommerce
# Cart Block: Refactor shipping totals. Separate cart and checkout logic. More clearly distinguish between Collection and Delivery

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Some parts of this were split into https://github.com/woocommerce/woocommerce/pull/52380. That PR should be merged first.

This refactor helps untangle the Shipping Totals component that is shared between the cart and checkout blocks. While they share the component, things like shipping calculator are unique to the cart page.

- Moves shared utility functions from the totals component into base utils
- Moves cart specific logic from from totals component into the cart shipping totals block. This is mainly because only the cart page needs a shipping calculator, but the checkout page uses the same component. I introduced a prop called `collaterals` to pass these extra components in.
- Adds a notice in cart if there are only local pickup options present and there is no shipping calculator available to use: "Delivery options will be calculated during checkout" This is because other shipping methods will appear at checkout.
- Made the label toggle between Delivery and Collection depending on your shipping method selection.
- Avoids prop drilling where possible and instead makes use of the `useStoreCart` hook.
- Updates tests.

### Screenshots

This is the display of collection vs delivery. Notice how the row label changes based on selection:

![Screenshot 2024-10-29 at 13 55 10](https://github.com/user-attachments/assets/2790b9c2-6f0c-41f3-a462-9f0e3c801fad)
![Screenshot 2024-10-29 at 13 55 35](https://github.com/user-attachments/assets/d07ff589-8f75-4ad2-9a18-61aab474de91)

I have tested:

- With shipping calc on and off
- Local pickup on and off
- With and without shipping rates
- With multiple packages

For context what brought this on, see p1729767564987079/1729685919.085089-slack-C02TS23QJ1X To test that use case:

- Create a zone called UK with flat rate
- Remove shipping methods from "everywhere" zone
- Enable local pickup
- Set default customer location to "no location"
- Disable the cart shipping calculator
- Enter cart page as a guest and see the shipping totals section. Note the wording "Delivery options will be calculated during checkout"

![Screenshot 2024-10-29 at 14 19 47](https://github.com/user-attachments/assets/154b3395-aa89-4ac1-9cf3-9ca4ff2c34ce)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a shipping zone called UK with flat rate
2. Remove shipping methods from "everywhere" zone
3. Enable local pickup
4. Disable the cart shipping calculator
5. Enter cart page as a guest and see the shipping totals section. Note the wording "Delivery options will be calculated during checkout"
6. As admin, go back and turn on the shipping calculator
7. As guest, confirm shipping calculator is displayed and functional.
8. Enter UK address.
9. Confirm delivery options appear.
10. Toggle between delivery/pickup and ensure labels etc update
11. Go to checkout. Confirm the selected shipping method is shown, and the calculator is not shown.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
